src/control/character_controller.rs

```diff
@@ -782,11 +782,40 @@ impl KinematicCharacterController {
         true
     }
 
-    /// For a given collision between a character and its environment, this method will apply
-    /// impulses to the rigid-bodies surrounding the character shape at the time of the collision.
+    /// For the given collisions between a character and its environment, this method will apply
+    /// impulses to the rigid-bodies surrounding the character shape at the time of the collisions.
     /// Note that the impulse calculation is only approximate as it is not based on a global
     /// constraints resolution scheme.
     pub fn solve_character_collision_impulses(
+        &self,
+        dt: Real,
+        bodies: &mut RigidBodySet,
+        colliders: &ColliderSet,
+        queries: &QueryPipeline,
+        character_shape: &dyn Shape,
+        character_mass: Real,
+        collisions: impl IntoIterator<Item = CharacterCollision>,
+        filter: QueryFilter,
+    ) {
+        for collision in collisions {
+            self.solve_character_collision_impulse(
+                dt,
+                bodies,
+                colliders,
+                queries,
+                character_shape,
+                character_mass,
+                &collision,
+                filter,
+            );
+        }
+    }
+
+    /// For the given collision between a character and its environment, this method will apply
+    /// impulses to the rigid-bodies surrounding the character shape at the time of the collision.
+    /// Note that the impulse calculation is only approximate as it is not based on a global
+    /// constraints resolution scheme.
+    fn solve_character_collision_impulse(
```

Comment:
Just to make it extra obvious it solves a single collision:
```suggestion
    fn solve_single_character_collision_impulse(
```

Reply:
Notice I changed its visibility to private, is that ok ? @sebcrozet

Reply:
Yes, making it private is good.